Issue 12747 update content type form

fmontes opened the pull request.

fmontes commented on src/app/portlets/content-types/edit/content-types-edit.component.html:
```diff
@@ -1,17 +1,15 @@
-<content-type-layout [contentTypeId]="data ? data.id : null">
-    <content-types-form
+<dot-content-type-layout [contentTypeId]="data?.id">
```
We don't need to do ```object && object.prop``` anymore :)

freddyucv replied:
cool

fmontes commented on src/app/portlets/content-types/edit/content-types-edit.component.html:
```diff
@@ -1,17 +1,15 @@
         #form
+        [fields]="fields"
         [data]="data"
-        [type]="contentTypeType"
```
Less inputs are always best.

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.ts:
```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
```
I think is better named it something like: content-types.resolver.ts, it is not really a service

fmontes replied:
I'm following the same pattern as with the guards as it is in the [angular documentation](https://angular.io/guide/router) they do like: ```src/app/crisis-center/crisis-detail-resolver.service.ts```

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.ts:
```diff
@@ -0,0 +1,43 @@
+import { DotRouterService } from '../../api/services/dot-router-service';
+
+@Injectable()
+export class ContentTypeResolver implements Resolve<ContentType> {
```
doc

fmontes replied:
Done

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.spec.ts:
```diff
@@ -0,0 +1,118 @@
+            .subscribe((fakeContentType: any) => {
+                expect(fakeContentType).toEqual({
+                    fake: 'content-type',
+                    object: 'right?'
```
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

fmontes replied:
https://github.com/dotCMS/core-web/pull/447#discussion_r144920884

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.spec.ts:
```diff
@@ -0,0 +1,118 @@
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((res: any) => {
+                expect(res).toBeNull();
```
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

fmontes replied:
So here is the deal, because the ```resolve``` is async also the method he use internally... but we are mocking all the responses and they are coming inmediatelly, we could use ```fakeAsync``` but it's not necessary because all the responses are inmediatelly available when the subscribe it's called.

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.spec.ts:
```diff
@@ -0,0 +1,118 @@
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((res: any) => {
+                expect(res).toEqual({
```
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

fmontes replied:
https://github.com/dotCMS/core-web/pull/447#discussion_r144920884

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.ts:
```diff
@@ -0,0 +1,43 @@
+
+    resolve(route: ActivatedRouteSnapshot): Observable<ContentType> {
+        if (route.paramMap.get('id')) {
+            return this.crudService
```
maybe we might change this for a private method something like getContentType

fmontes replied:
Yup, great this way we can have a clear code.

freddyucv commented on src/app/portlets/content-types/content-types-resolver.service.ts:
```diff
@@ -0,0 +1,43 @@
+                });
+        } else {
+            const type = route.paramMap.get('type').toUpperCase();
+            return Observable.of(
```
maybe we might change this for a private method something like return defaultContentType

fmontes replied:
Yup, great this way we can have a clear code.

freddyucv commented on src/app/portlets/content-types/edit/content-types-edit.component.ts:
```diff
@@ -22,120 +17,102 @@ import { DotConfirmationService } from '../../../api/services/dot-confirmation';
-    private handleFormSubmissionResponse(res: any): void {
-        this.form.resetForm();
+    private getCreateContentTypeData(value: any): ContentType {
+        return Object.assign({}, CONTENT_TYPE_INITIAL_DATA, value);
```
is this really necessary? isn't this do by the resolver?

fmontes replied:
I change this, now the form control handle ALL the Content Type object even when some properties are not represent for actual fields in the HTML

hmoreras commented:
Feedback after QA: 

- When expanded edit button should be disabled - is not happening.
- Openthe form change something, Update, then open the form agin and the “Update” btn keep enable. Should be disable since nothing change since the last Update.
- while doing tab to change between fields, should be nice to ignore the disabled fields. 
- Cancel Label not loading.. can be my environment. 
- When fields are empty is hard to identify witch ones are enable/disable.

fmontes commented:
- When expanded edit button should be disabled - is not happening.
Done
- Open the form change something, Update, then open the form agin and the “Update” btn keep enable. Should be disable since nothing change since the last Update.
Done

- while doing tab to change between fields, should be nice to ignore the disabled fields.
I'll create an issue for this

- Cancel Label not loading.. can be my environment.
Done

- When fields are empty is hard to identify witch ones are enable/disable.
Need to work on this, maybe change the design to be a dotted line
